annotation_widgets/event_validation/gui.py

```diff
@@ -0,0 +1,183 @@
+        sep9 = ttk.Separator(self, orient='vertical')
+        sep9.grid(row=0, column=11, sticky='ns')
+
+        self.preview_mode_label.grid(row=0, column=12, sticky='ew', padx=15)
+        sep10 = ttk.Separator(self, orient='vertical')
+        sep10.grid(row=0, column=13, sticky='ns')
+
+        # Video Frames info label (initially hidden)
+        self.frame_info_label.grid(row=0, column=14, sticky='ew', padx=15)
+
+        self.columnconfigure(8, weight=1)  # Make progress bar expand
+
+    def on_resize(self, event):
+        # Calculate an appropriate font size based on the current width
+        new_font_size = max(8, min(15, int(self.winfo_width() / 130)))
+        label_font = font.Font(family="Ubuntu Condensed", size=new_font_size)
+
+        # Set the new font to all labels and progress bar
+        for widget in [self.mode_label, self.item_id_label, self.speed_label, self.processed_label, self.duration_label,
+                       self.preview_mode_label, self.frame_info_label]:
+            widget.config(font=label_font)
+
+    def update_status(self):
+        status_data = self.logic.status_data
+
+        # Update labels
+        self.mode_label.config(text=f"Mode: Event Validation")
+        self.item_id_label.config(text=f"Item id: {status_data.item_id}")
+        self.speed_label.config(text=f"Speed: {status_data.speed_per_hour} img/hour")
+
+        position_percent = int((status_data.item_id + 1) / status_data.number_of_items * 100)
+        self.processed_label.config(text=f"Position: {position_percent} % ({status_data.item_id + 1}/{status_data.number_of_items})")
+        self.progress_bar["value"] = position_percent
+        self.duration_label.config(text=f"Duration: {status_data.annotation_hours} hours")
+
+        self.preview_mode_label.config(text=f"Preview mode: {self.logic.view_mode}")
+
+        if self.logic.video_mode:
+            self.frame_info_label.config(text=f"Frame info: {self.logic.current_frame_number + 1} / {self.logic.number_of_frames}")
+            self.frame_info_label.grid()
+        else:
+            self.frame_info_label.grid_remove()
+
+        # Schedule the next update
+        self.after(10, self.update_status)
+
+
+class EventValidationSideBar(tk.Frame):
+    def __init__(self, parent, logic, *args, **kwargs):
+        super().__init__(parent, *args, **kwargs)
+        self.logic = logic
+        self.parent = parent
+
+        self.question_frames = []
+        self.answer_vars = OrderedDict()
+        self.answer_buttons = OrderedDict()
+
+        self.logic.on_item_change(self.update_display)
+        self.create_question_widgets()
+
+        self.comment_label = tk.Label(self, text="Comment")
+        self.comment_label.pack(anchor="w", padx=10, pady=2)
+        self.comment_entry = tk.Text(self, wrap="word", height=2, width=40)
+        self.comment_entry.pack(anchor="w", padx=10, pady=2)
+
+        self.comment_entry.bind("<Escape>", self.save_comment)
+
```

Comment:
Show the number on keyboard coresponding to the question
![image](https://github.com/user-attachments/assets/2f7c487e-3686-4c89-aa66-1080e22e683c)
In this format:
```
[1] TP/FP
...
[2] Is person...
```

Reply:
Done